0000-route-broadcasting-protocol/RouteBroadcastingProtocol.asn

```diff
@@ -0,0 +1,60 @@
+RouteBroadcastingProtocol DEFINITIONS AUTOMATIC TAGS ::= BEGIN
+
+IMPORTS
+  UInt8,
+  UInt16,
+  UInt32
+  FROM GenericTypes
+  Address
+  FROM InterledgerTypes
+;
+
+-- Route Control Request
+
+SyncMode ::= INTEGER {
+  idle (0),
+  sync (1)
+} (0..255)
+
+RoutingTableId ::= OCTET STRING (SIZE (16))
+
+Feature ::= OCTET STRING
+
+RouteControlRequestData ::= SEQUENCE {
+  syncMode SyncMode, -- the state that the node wants the counterpart node to be
+  routingTableId RoutingTableId, -- the routing table ID that the requesting node knows at that point
+  knownEpochIndex UInt32, -- the epoch that the requesting node knows at that point
+  features SEQUENCE OF Feature -- reserved for the future, currently not used
+}
+
+-- Route Update Request
+
+Auth ::= OCTET STRING (SIZE (32))
+
+Property ::= SEQUENCE {
+  meta UInt8,
+  id UInt16,
+  value OCTET STRING
+}
+
+NewRoute ::= SEQUENCE {
+  addedRoutePrefix Address, -- the prefix of the added route
+  path SEQUENCE OF Address, -- the hops destinated to the prefix node
+  auth Auth, -- reserved for the future, currently not used
+  props SEQUENCE OF Property -- reserved for the future, currently not used
+}
+
+-- NOTICE: The log of "toEpochIndex" will NOT be included.
+-- e.g. with this log: [ a, b, c, d ], update request from 0 to 2 means [ a, b ]
+RouteUpdateRequestData ::= SEQUENCE {
+  routingTableId RoutingTableId, -- the routing table ID that the requesting node knows at that point
+  currentEpochIndex UInt32, -- current epoch index of the node that requests the update for the other nodes
+  fromEpochIndex UInt32, -- the epoch index that the log of this update request starts from
+  toEpochIndex UInt32, -- the epoch index that the log of this update request ends at
```

Comment:
I assume this is `inclusive` (as opposed to `exclusive`), but might be worth stating directly. For example, if `fromEpoch=1`, and `toEpoch=3`, then I would expect 3 routes (1, 2, and 3).

Reply:
I referred to this code:
https://github.com/interledgerjs/ilp-connector/blob/master/src/routing/ccp-sender.ts#L182-L186
and this specification:
https://www.ecma-international.org/ecma-262/5.1/#sec-15.4.4.10

As a conclusion, in my opinion, `3` is not included in your case. See the picture below.
![image](https://user-images.githubusercontent.com/811594/44450511-1b348a00-a62c-11e8-8591-ab598c87ca57.png)

Am I wrong?

Reply:
Yes, I think you're right. In that case, we should use the term `exclusive`, like this:

`the epoch index that the log of this update request ends at, exclusive`